src/shell/Shell.cpp

```diff
@@ -795,6 +795,84 @@ static void executeWAST(Store* store, const std::string& filename, const std::ve
     }
 }
 
+static void runExports(Store* store, const std::string& filename, const std::vector<uint8_t>& src, std::string& exportToRun)
+{
+    auto parseResult = WASMParser::parseBinary(store, filename, src.data(), src.size());
+
+    if (!parseResult.second.empty()) {
+        fprintf(stderr, "parse error: %s\n", parseResult.second.c_str());
+        return;
+    }
+
+    auto module = parseResult.first;
+
+    const auto& importTypes = module->imports();
+
+    ExternVector importValues;
+
+    if (importTypes.size() != 0) {
+        fprintf(stderr, "error: module has imports, but imports are not supported\n");
+        return;
+    }
+
+    struct RunData {
+        Module* module;
+        ExternVector& importValues;
+        ExportType* exp;
+    } data = { module.value(), importValues, nullptr };
+    Walrus::Trap trap;
+
+    for (auto&& exp : module->exports()) {
+        if (exp->exportType() == ExportType::Function) {
+            if (exportToRun != exp->name() && exportToRun != "*") {
+                continue;
+            }
+
+            data.exp = exp;
+            trap.run([](ExecutionState& state, void* d) {
+                auto data = reinterpret_cast<RunData*>(d);
+                Instance* instance = data->module->instantiate(state, data->importValues);
```

Comment:
In this way, `instantiate` creates a new `Instance` object everytime when an export function is called.
Instead, what about creating an `Instance` object once and reusing it for each call?